.github/workflows/codeql-analysis.yaml

```diff
@@ -0,0 +1,51 @@
+---
+name: "CodeQL"
+
+on:
+  push:
+    branches: [main]
+    paths-ignore:
+      - '**.md'  # All markdown files in the repository
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [main]
+    paths-ignore:
+      - '**.md'
+  schedule:
+    - cron: '43 22 * * 2'
```

Comment:
At 10:43PM on Tuesdays -- Why? If it's just to have a weekly run that's niceish to GitHubs cron jobs, perhaps a comment to that effect?